tests/src/topology/mod.rs

```diff
@@ -255,4 +261,287 @@ impl Topology {
     pub fn executors(&self) -> &[Executor] {
         &self.executors
     }
+
+    pub async fn wait_network_ready(&self) {
+        const POLL_INTERVAL: Duration = Duration::from_millis(200);
+        const TIMEOUT: Duration = Duration::from_secs(60);
+
+        let listen_ports = self.node_listen_ports();
+        if listen_ports.len() <= 1 {
+            return;
+        }
+
+        let initial_peer_ports = self.node_initial_peer_ports();
+        let expected_peer_counts = find_expected_peer_counts(&listen_ports, &initial_peer_ports);
+        let labels = self.node_labels();
+
+        let readiness = timeout(adjust_timeout(TIMEOUT), async {
+            loop {
+                let validator_infos =
+                    join_all(self.validators.iter().map(Validator::network_info)).await;
+
+                let executor_infos =
+                    join_all(self.executors.iter().map(Executor::network_info)).await;
```

Comment:
As this is async, you can just get both futures and await both of them at the same time.